### PR TITLE
UFO Secondary Indexes remove comparator for indexes on Protobufs

### PR DESCRIPTION
**Pull request by zhangn49**
## Overview

Description:

Why should this be merged: 

Related issue(s) (if applicable): #<number>

## Checklist (Definition of Done):

- [x] There are no TODOs left in the code
- [x] [Coding conventions](https://github.com/CorfuDB/CorfuDB/wiki/Corfu-Style-Guidelines) (e.g. for logging, unit tests) have been followed
- [x] Change is covered by automated tests
- [x] Public API has Javadoc

### CODE REVIEW

File: test/src/test/resources/proto/sample_schema.proto [hunk 1]
```diff
@@ -41,8 +41,7 @@ message EventInfo {
     optional uint32 port = 3;
     optional int64 event_time = 4 [(org.corfudb.runtime.schema).secondary_key = true];
     optional uint32 frequency = 5;
-    // TODO: Need to make the following work.. looking for ideas..
-    // optional Uuid uuid = 6 [(org.corfudb.runtime.schema).secondary_key = true];
+    optional Uuid uuid = 6 [(org.corfudb.runtime.schema).secondary_key = true];
```
Comment (hisundar) [inline test/src/test/resources/proto/sample_schema.proto:44]: this protobuf isn't used in the current CorfuShimStoreTest.java
Could you please help change the ExampleValue inside example_schema.proto in the runtime directory instead? That way we will instantly know if it works.

**Review by hisundar — CHANGES_REQUESTED** [review]
Thanks for the change Nan, just one comment on the test coverage for this.

File: test/src/test/resources/proto/sample_schema.proto (hunk 1 above)
Comment (zhangn49) [reply test/src/test/resources/proto/sample_schema.proto:44]: Done.

**Review by hisundar — APPROVED** [review]
LGTM

File: runtime/src/main/java/org/corfudb/runtime/collections/CorfuTable.java [hunk 2]
```diff
@@ -78,18 +78,18 @@
 
     // The "main" map which contains the primary key-value mappings.
     private final ContextAwareMap<K, V> mainMap;
-    private final Set<Index.Spec<K, V, ? extends Comparable>> indexSpec;
-    private final Map<String, Map<Comparable, Map<K, V>>> secondaryIndexes;
+    private final Set<Index.Spec<K, V, ?>> indexSpec;
```
Comment (annym) [inline runtime/src/main/java/org/corfudb/runtime/collections/CorfuTable.java:81]: missing getByIndex and getByIndexAndFilter + unit test

Comment (zhangn49) [reply runtime/src/main/java/org/corfudb/runtime/collections/CorfuTable.java:81]: Done

Comment (zhangn49) [reply runtime/src/main/java/org/corfudb/runtime/collections/CorfuTable.java:81]: Done

File: test/src/test/java/org/corfudb/runtime/collections/CorfuTableTest.java [hunk 3]
```diff
@@ -114,6 +116,40 @@ public void cannotLookupByIndexAndFilterWhenIndexNotSpecified() {
         corfuTable.getByIndexAndFilter(StringIndexer.BY_FIRST_LETTER, p -> p.getValue().contains("cd"), "a");
     }
 
+    /**
+     * Verify that a lookup by indexAndFiler works when index is a
```
Comment (annym) [inline test/src/test/java/org/corfudb/runtime/collections/CorfuTableTest.java:120]: typo -> indexAndFiler

**Review by annym — APPROVED** [review]
LGTM!

**Pull request merged**